### ISSUE TITLE
problems computing dtt on Mac OSx. 

### ISSUE THREAD

**Issue by echavess** [ROBelgium/MSNoise#209]
I still have the same issue in Mac OSX. Catalina every time I restart the machine or Mysql. 
I re-installed mysql and msnoise and still get this problem:

```
sqlalchemy.exc.InternalError: (pymysql.err.InternalError) (1055, "Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'msnoise_sses.jobs.ref' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by")
[SQL: SELECT jobs.ref AS jobs_ref, jobs.day AS jobs_day, jobs.pair AS jobs_pair, jobs.jobtype AS jobs_jobtype, jobs.flag AS jobs_flag, jobs.lastmod AS jobs_lastmod
FROM jobs
WHERE jobs.day >= %(day_1)s AND jobs.day <= %(day_2)s AND jobs.jobtype = %(jobtype_1)s AND jobs.lastmod >= %(lastmod_1)s GROUP BY jobs.day ORDER BY jobs.day]
[parameters: {'day_1': datetime.date(2017, 1, 1), 'day_2': datetime.date(2017, 12, 31), 'jobtype_1': 'DTT', 'lastmod_1': datetime.datetime(2020, 8, 2, 14, 44, 7, 483859)}]
(Background on this error at: http://sqlalche.me/e/13/2j85)
```

The fix for Linux doesn't work in Mac, which is a frustrating. I found this hack on[ stackoverflow](https://stackoverflow.com/questions/41887460/select-list-is-not-in-group-by-clause-and-contains-nonaggregated-column-inc): 

`SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''));`

So I type that on MySQLWorkbench: 

<img width="1435" alt="Screen Shot 2020-08-03 at 14 49 46" src="https://user-images.githubusercontent.com/5248851/89226119-a482c400-d598-11ea-94fa-9eff0d644f31.png">

The problems goes away but I get a new problem: 

```
 ech@Esteban  ~/Work/CentralPacific_repeaters/Data_processing/Noise  msnoise compute_dtt                       1 ↵  10478  14:44:13
/Users/ech/anaconda3/lib/python3.7/site-packages/obspy/signal/headers.py:93: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  ], align=True)
2020-08-03 20:51:17.791486 msnoise [pid 2070][INFO]: *** Starting: Compute DT/T ***
2020-08-03 20:51:17.929510 msnoise [pid 2070][INFO]: Loading mov=5 days for filter=01
Traceback (most recent call last):
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/series.py", line 1002, in __setitem__
    self._set_with_engine(key, value)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/series.py", line 1033, in _set_with_engine
    loc = self.index._engine.get_loc(key)
  File "pandas/_libs/index.pyx", line 70, in pandas._libs.index.IndexEngine.get_loc
  File "pandas/_libs/index.pyx", line 75, in pandas._libs.index.IndexEngine.get_loc
TypeError: '[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 23 31 32 33 34 35 36 37
 38 39 40 41 42 43 44 45 46]' is an invalid key

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/ech/anaconda3/bin/msnoise", line 33, in <module>
    sys.exit(load_entry_point('msnoise==1.6', 'console_scripts', 'msnoise')())
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/msnoise/scripts/msnoise.py", line 1252, in run
    cli(obj={})
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/msnoise/scripts/msnoise.py", line 824, in compute_dtt
    main(loglevel=loglevel)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/msnoise/s06compute_dtt.py", line 231, in main
    df['err'][tmp] = 1.0
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/series.py", line 1026, in __setitem__
    self._set_with(key, value)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/series.py", line 1061, in _set_with
    self.loc[key] = value
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 666, in __setitem__
    indexer = self._get_setitem_indexer(key)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 617, in _get_setitem_indexer
    return self._convert_to_indexer(key, axis=0, is_setter=True)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 1198, in _convert_to_indexer
    return self._get_listlike_indexer(key, axis, raise_missing=True)[1]
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 1254, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis, raise_missing=raise_missing)
  File "/Users/ech/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 1304, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")
KeyError: '[1, 2, 3, 4, 6, 7, 8, 9, 11, 12, 13, 14, 23, 31, 32, 33, 34, 36, 37, 38, 39, 41, 42, 43, 44, 46] not in index'
```

ideas?

_Originally posted by @echavess in https://github.com/ROBelgium/MSNoise/issues/181#issuecomment-668236090_

**Comment by ThomasLecocq** [comment]
looks like a pandas-related change.. will have to run the tests with the latest versions (nomally Travis does it, will check)

**Comment by ThomasLecocq** [comment]
solved with a8288e0a4d3dc9c2b247c78b95445a752cb07292
